boards: arm: gd32f450z_eval: add initial support

sbourdelin opened the pull request:
Add initial support for the GD32F450Z-EVAL board.

Signed-off-by: Sebastien Bourdelin <sebastien.bourdelin@rtone.fr>

sbourdelin committed:
boards: arm: gd32f450z_eval: add initial support

Add initial support for the GD32F450Z-EVAL board.

Signed-off-by: Sebastien Bourdelin <sebastien.bourdelin@rtone.fr>

gmarull commented on boards/arm/gd32f450z_eval/gd32f450z_eval.dts:
```diff
@@ -0,0 +1,155 @@
+	status = "okay";
+};
+
+&gpioe {
+	status = "okay";
+};
```
this one doesn't seem to be used

sbourdelin replied:
indeed, thx!

gmarull commented on boards/arm/gd32f450z_eval/gd32f450z_eval.dts:
```diff
@@ -0,0 +1,155 @@
+	status = "okay";
+};
+
+&gpiof {
+	status = "okay";
+};
```
same here

sbourdelin replied:
ditto

gmarull commented on boards/arm/gd32f450z_eval/gd32f450z_eval.yaml:
```diff
@@ -0,0 +1,15 @@
+  - zephyr
+  - gnuarmemb
+  - xtools
+supported:
+  - pwm
```
this is missing a few entries (I see other boards should be fixed as well)
- gpio
- dac
- spi
- i2c

this is used by the testing infrastructure

sbourdelin replied:
yes thx, i should probably add `uart` too

gmarull reviewed (commented):
lgtm, a few minor nitpicks

nandojve commented on boards/arm/gd32f450z_eval/support/openocd.cfg:
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2022, Rtone.
```
Don`t forget to keep rights if you copy and adapted this part.

```diff
+# Copyright (c) 2021, ATL-Electronics
+# Copyright (c) 2021, Teslabs Engineering S.L.
# Copyright (c) 2022, Rtone.
```

sbourdelin replied:
oupsy, got it, thx!

sbourdelin commented:
Hi @nandojve ,

No indeed, i m gonna take a look

sbourdelin commented:
Hi @nandojve ,

So, if i'm not misunderstanding, the problem is that the `gigadevice HAL` already use the `SPI_SLAVE` define name for the register configuration [[1](https://github.com/zephyrproject-rtos/hal_gigadevice/blob/main/gd32f4xx/standard_peripheral/include/gd32f4xx_spi.h#L162)] where the `spi_looback` test driver use the same define name to define which slave should be addressed [[2](https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/spi/spi_loopback/src/spi.c#L20)].

Not sure how you want to address this case?
As any other Gigadevice boards using this HAL will suffer from the same problem, i was thinking we could rename the define  `SPI_SLAVE` in the  `spi_looback` to `SPI_SLAVE_NUMBER` which might actually looks more accurate too.
What do you think?

gmarull commented:
> Hi @nandojve ,
> 
> So, if i'm not misunderstanding, the problem is that the `gigadevice HAL` already use the `SPI_SLAVE` define name for the register configuration [[1](https://github.com/zephyrproject-rtos/hal_gigadevice/blob/main/gd32f4xx/standard_peripheral/include/gd32f4xx_spi.h?rgh-link-date=2022-05-11T14%3A48%3A48Z#L162)] where the `spi_looback` test driver use the same define name to define which slave should be addressed [[2](https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/spi/spi_loopback/src/spi.c?rgh-link-date=2022-05-11T14%3A48%3A48Z#L20)].
> 
> Not sure how you want to address this case? As any other Gigadevice boards using this HAL will suffer from the same problem, i was thinking we could rename the define `SPI_SLAVE` in the `spi_looback` to `SPI_SLAVE_NUMBER` which might actually looks more accurate too. What do you think?

makes sense for a quick solution, also agree with @nandojve suggestion.

sbourdelin commented:
> However, it will be important open an enhancement to remove all the recursive dependencies related to `hal_gigadevice/<CONFIG_SOC_SERIES>/standard_peripheral)/include/<CONFIG_SOC_SERIES>_libopt.h`. Then link that issue with this PR to allow track the origin.

I'm sorry you lost me there, i'm not sure if you want me to do something about it and i'm even less sure to see the link between the redefine and the recursive dependencies 😅

nandojve commented:
Hi @sbourdelin ,

No problem, let's try clarify.

> I'm sorry you lost me there, i'm not sure if you want me to do something about it and i'm even less sure to see the link between the redefine and the recursive dependencies sweat_smile

You need to do 2 things:
- Add your proposal to change SPI_SLAVE to SPI_SLAVE_NUMBER_NUMBER to fix CI and get this merged.
- Open an Issue to remind maintainers that there is an issue related to hal_gigadevice/<CONFIG_SOC_SERIES>/standard_peripheral)/include/<CONFIG_SOC_SERIES>_libopt.h.

sbourdelin committed:
tests: spi: spi_loopback: rename SPI_SLAVE to SPI_SLAVE_NUMBER

As the SPI_SLAVE define is used by some HAL (e.g. the gigadevice HAL),
SPI_SLAVE has been renamed to SPI_SLAVE_NUMBER.

Signed-off-by: Sebastien Bourdelin <sebastien.bourdelin@rtone.fr>

sbourdelin commented:
hi @nandojve 

Thanks for the explanations, i added a commit to this PR for the spi test, maybe you want it not link to this PR?
For your suggestion on the Gigadevice HAL, it doesn't seems i can open an issue on the https://github.com/zephyrproject-rtos/hal_gigadevice repository.

Let me know, thanks!

nandojve commented on tests/drivers/spi/spi_loopback/src/spi.c:
```diff
@@ -17,7 +17,7 @@ LOG_MODULE_REGISTER(main);
 #include <drivers/spi.h>
 
 #define SPI_DRV_NAME	CONFIG_SPI_LOOPBACK_DRV_NAME
-#define SPI_SLAVE	CONFIG_SPI_LOOPBACK_SLAVE_NUMBER
+#define SPI_SLAVE_NUMBER	CONFIG_SPI_LOOPBACK_SLAVE_NUMBER
 #define SLOW_FREQ	CONFIG_SPI_LOOPBACK_SLOW_FREQ
 #define FAST_FREQ	CONFIG_SPI_LOOPBACK_FAST_FREQ
```
Please keep vertikal alignment for CONFIG_*

sbourdelin replied:
sure! done

nandojve commented:
Hi @sbourdelin ,

> For your suggestion on the Gigadevice HAL, it doesn't seems i can open an issue on the https://github.com/zephyrproject-rtos/hal_gigadevice repository.

Please open issue here on ZephyrProject and you can assign to me.

sbourdelin commented:
Hi @nandojve ,

> Hi @sbourdelin ,
> 
> > For your suggestion on the Gigadevice HAL, it doesn't seems i can open an issue on the https://github.com/zephyrproject-rtos/hal_gigadevice repository.
> 
> Please open issue here on ZephyrProject and you can assign to me.

Issue is here: https://github.com/zephyrproject-rtos/zephyr/issues/45720
It looks i can't assign it to you directly.

gmarull commented on boards/arm/gd32f450z_eval/Kconfig:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022, Rtone.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_INIT_PRIORITY
+	int "Board initialization priority"
+	default 50
+	help
+	  Board initialization priority.
```
where is this used btw?

sbourdelin replied:
This is indeed not required, lazy copy paste from the GD32F450I board, my bad.
I'm deleting it thx!
